### DISCUSSION TITLE
Will data be pushed or do I need to pull?

### DISCUSSION THREAD

**Discussion by wizzywik** [tillsteinbach/WeConnect-mqtt#67] [Q&A] [answered]
Installed WeConnect-mqtt and configured a MQTT broker. Do I need to pull data from this client or is it pushed by the client.
(Probably a stupid question, but I'm not that familiar with MQTT and programming in general)

**Comment by tillsteinbach** [comment]
It is pushed from WeConnect-mqtt to your broker. In your client you have to subscribe. Subscribing to `#` gives you a subscription to everything.

**Comment by boredomwontgetus** [comment, marked as answer]
As i have some time left i take the liberty to answer for Till! (He will correct me if i am wrong)

1.) weconnect-mqtt polls the VW servers in a certain interval (i believe 300s per default)
2.) as soon as it receives data from the VW servers it publishes (push) it to your mqtt-broker
3.) from there you can grab (subscribe) it with any mqtt-client you wish. this way your mqtt-client sees new values anytime they are updated. 

see also: https://en.wikipedia.org/wiki/Publish%E2%80%93subscribe_pattern

HTH, Tom

**Comment by wizzywik** [comment]
Thanks guys! Will try it again and let you know.
As a little background: I'm planning to use WeConnect-mgtt to integrate to my home-automation (Loxone) to implement local smart charging (solar). VwsFriends with Grafana is nice as a stand-alone monitoring tool. Hope this makes sense.

**Comment by wizzywik** [comment]
Subscription was the step I missed! Thnx again!
